adyenv6b2ccheckoutaddon/acceleratoraddon/web/webroot/_ui/responsive/common/js/adyen.checkout.js

```diff
@@ -101,7 +103,22 @@ var AdyenCheckoutHybris = (function () {
                 }
             }
 
-            $( 'input[name="txvariant"]' ).remove();
+            if( paymentMethod === "sepadirectdebit" ){
+                if( !this.sepaDirectDebit.state.isValid ) {
+                    window.alert("Invalid SEPA Owner Name and IBAN number");
+                    return false;
+                }
+                var sepaOwnerNameField = document.getElementById('sepaOwnerName');
+                var sepaIbanNumberField = document.getElementById('sepaIbanNumber');
+                if( sepaOwnerNameField.value === "" ) {
+                    window.alert("Please enter Owner Name");
+                    return false;
+                }
+                if( sepaIbanNumberField.value === "" ) {
+                    window.alert("Please enter IBAN Number");
+                    return false;
```

Comment:
if one of the values is empty, the function should return already in 109 since the state is not valid, right? maybe we can remove these two ifs?

Reply:
good point.